docs/preview/02-Features/publishing-events.md

```diff
@@ -29,227 +29,312 @@ public void ConfigureServices(IServiceCollection services)
+    // For Arcus HTTP correlation, see: https://webapi.arcus-azure.net/features/correlation
+    services.AddHttpCorrelation();
 
-```csharp
-using Arcus.EventGrid.Publishing;
-
-IEventGridPublisher publisher = 
-    EventGridPublisherBuilder.ForTopic("https://my-eventgrid-topic-endpoint")
-                             .UsingAuthenticationKey("<my-authentication-secret>")
-                             .Build();
+    // Registers an `EventGridPublisherClient` to your application to a custom topic.
+    services.AddAzureClients(clients =>
+    {
+        clients.AddEventGridPublisherClient(
+            // Custom Azure Even Grid topic endpoint:
+            "https://my-eventgrid-topic-endpoint", 
+            // Secret name where the authentication key to interact with Azure Event Grid is stored in the Arcus secret store.
+            "<my-authentication-secret-name>");
+    });
+}
 ```
 
 ## Publishing events
+Publishing events has no difference than only using Microsoft's `EventGridPublisherClient`, as the correlation tracking happens internally.
 
-### Publishing EventGridEvents
-The library provides support to publish custom Azure EventGrid events based on your custom definition of an event, to your custom Azure EventGrid topic endpoint. In this case, `NewCarRegistered` is the created, which inherits `Event`:
-
+To publish events, inject the client and choose between publishing cloud events, Event Grid events or custom events.
 ```csharp
-public class NewCarRegistered : EventGridEvent<CarEventData>
-{
-     private const string DefaultDataVersion = "1", 
-                          DefaultEventType = "Arcus.Samples.Cars.NewCarRegistered";
+using Azure.Messaging.EventGrid;
+using Microsoft.Extensions.Azure;
 
-    // Usually required for serialization.
-    private NewCarRegistered()
+public class MyService
+{
+    public MyService(IAzureClientFactory<EventGridPublisherClient> clientFactory)
     {
-    }
+        EventGridPublisherClient client = clientFactory.CreateClient("Default");
 
-    public NewCarRegistered(string id, string subject, string licensePlate) 
-        : base(id, subject: "New registered car", new CarEventData(licensePlate), DefaultDataVersion, DefaultEventType) 
-    {
+        client.SendEventAsync(new CloudEvent(...));
+
+        client.SendEventAsync(new EventGridEvent(...));
+
+        // And other overloads...
     }
 }
 ```
 
-Once the event definition is ready, you can publish it with the previously created `IEventGridPublisher` implementation:
+For more information on Azure core clients, how they work and how to use them, see [Microsoft's documentation](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/microsoft.extensions.azure-readme-pre).
```

Comment:
What is meant with Azure core clients ?

Reply:
The clients registered with the `AddAzureClients`. Their also exists a method called `AddAzureClientsCore` which only provides the minimum services needed.
Yeah, maybe this is a bit confusing and should just be 'Azure clients', right?

Reply:
Changed in https://github.com/arcus-azure/arcus.eventgrid/pull/245/commits/6bd28b4996f6f1da3f97f94514d225c228a5f500